pkg/kubectl/cmd/set/set_resources.go

```diff
@@ -0,0 +1,233 @@
+	Resources              []string
+}
+
+const (
+	resources_long = `Update existing container resource limits/requests. Setting limits/requests restricts cpu/memory usage on 
+running containers, which is normally unbound. A container with limits/requests set changes the QoS tiers
+of the container above those without limits/requests, this can be most important for infrastructure pods. 
+Pods with set limits/requests will fail to  deploy if a node that can handle the limits/requests cannot 
+be found.  
+
+*NOTE* if limits are set without requests the requests are set to the limits.
+
+Possible resources include (case insensitive):`
+
+	resources_example = `
+# Set a deployments nginx container cpu limits to "200m and memory to "512Mi"
+
+kubectl set resources deployment nginx -c=nginx --limits=cpu=200m,memory=512Mi
+
+# Set the resources and requests for all containers in nginx
+
+kubectl set resources deployment nginx --limits=cpu=200m,memory=512Mi --requests=cpu=100m,memory=256Mi
+
+# Print the result (in yaml format) of updating nginx container limits from a local, without hitting the server
+
+kubectl set resources -f path/to/file.yaml --limits=cpu=200m,memory=512Mi --local -o yaml
+`
+)
+
+func NewCmdResources(f *cmdutil.Factory, out io.Writer, errOut io.Writer) *cobra.Command {
+	options := &ResourcesOptions{
+		Out: out,
+		Err: errOut,
+	}
+	var pod_specs string
+	RESTMappings := f.ResourcesWithPodSpecs()
+	for _, Map := range RESTMappings {
+		pod_specs = pod_specs + ", " + Map.Resource
+
+	}
+
+	cmd := &cobra.Command{
+		Use:     "resources (-f FILENAME | TYPE NAME)  ([--limits=LIMITS & --requests=REQUESTS] ^ --remove)",
+		Short:   "update resource limits/requests on objects with pod templates",
+		Long:    resources_long + "\n" + pod_specs[2:],
+		Example: resources_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "Filename, directory, or URL to a file identifying the resource to get from the server"
+	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)
+	cmd.Flags().BoolVar(&options.All, "all", false, "select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on")
+	cmd.Flags().StringVarP(&options.ContainerSelector, "containers", "c", "*", "The names of containers in the selected pod templates to change, all containers are selected by default - may use wildcards")
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set resources will NOT contact api-server but run locally")
+	cmdutil.AddRecordFlag(cmd)
+	cmd.Flags().BoolVar(&options.Remove, "remove", false, "If true, set resources will remove resource limit/requests from pod template")
```

Comment:
need to read code more, but this seems like it warrants an example.  does it remove both requests and limits, or just one, or just for a specific resource?  why do i want this in practice?  this seems error prone as you are basically making it relatively easy to go from guaranteed qos to best-effort, when a user could just say --requests=cpu=0,memory=0 --limits=cpu=0,memory=0

nit: typically, we refer to this as requests/limits and not limit/request, will abstain from commenting the same throughout the PR, but please reverse where appropriate.